chips/msp432/src/cs.rs

```diff
@@ -0,0 +1,269 @@
+        ACLK_READY OFFSET(24) NUMBITS(1),
+        MCLK_READY OFFSET(25) NUMBITS(1),
+        HSMCLK_READY OFFSET(26) NUMBITS(1),
+        SMCLK_READY OFFSET(27) NUMBITS(1),
+        BCLK_READY OFFSET(28) NUMBITS(1)
+    ],
+    // interrupt enable register
+    CSIE [
+        // LFXT oscillator fault flag
+        LFXTIE OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIE OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIE OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIE OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIE OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt flag register
+    CSIFG [
+        // LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt clear register
+    CSCLRIFG [
+        // LFXT oscillator fault flag
+        LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // interrupt set/assert register
+    CSSETIFG [
+        // LFXT oscillator fault flag
+        SET_LFXTIFG OFFSET(0) NUMBITS(1),
+        // HFXT oscillator fault flag
+        SET_HFXTIFG OFFSET(1) NUMBITS(1),
+        // DCO external resistor open circuit fault flag
+        SET_DCOR_OPNIFG OFFSET(6) NUMBITS(1),
+        // LFXT start fault counter
+        SET_FCNTLFIFG OFFSET(8) NUMBITS(1),
+        // HFXT start fault counter
+        SET_FCNTHFIFG OFFSET(9) NUMBITS(1)
+    ],
+    // DCO external resistor calibration 0 register
+    CSDCOERCAL0 [
+        // DCO temperature compensation calibration
+        DCO_TCCAL OFFSET(0) NUMBITS(1),
+        // DCO frequency calibration for DCO frequency range (DCORSEL) 0 to 4
+        DCO_FCAL_RSEL04 OFFSET(16) NUMBITS(10)
+    ],
+     // DCO external resistor calibration 1 register
+    CSDCOERCAL1 [
+        // DCO frequency calibration for DCO frequency range (DCORSEL) 5
+        DCO_FCAL_RSEL5 OFFSET(0) NUMBITS(10)
+    ]
+];
+
+#[allow(dead_code)]
+#[repr(u32)]
+enum DcoFrequency {
+    _1_5Mhz = 0,
+    _3Mhz = 1,
+    _6Mhz = 2,
+    _12Mhz = 3,
+    _24Mhz = 4,
+    _48Mhz = 5,
+}
+
+pub struct ClockSystem {
+    registers: StaticRef<CsRegisters>,
+}
+
+impl ClockSystem {
+    pub const fn new() -> ClockSystem {
+        ClockSystem { registers: CS_BASE }
+    }
+
+    #[inline]
```

Comment:
You probably can just let the compiler inline this, I don't think you need it to be explicit.